temporal/server_test.go

```diff
@@ -37,16 +43,93 @@ import (
 	"go.temporal.io/server/tests/testutils"
 )
 
-// TestNewServer verifies that NewServer doesn't cause any fx errors
+// TestNewServer verifies that NewServer doesn't cause any fx errors, and that there are no unexpected error logs after
+// running for a few seconds.
 func TestNewServer(t *testing.T) {
+	t.Parallel()
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
```

Comment:
this call is not needed anymore

Reply:
https://pkg.go.dev/github.com/golang/mock/gomock#Controller.Finish

Reply:
oh nice, I learned that at some point but keep forgetting it

Reply:
Yeah, I added it per @dnr 's suggestion.

Reply:
Removed